ClientTests/WallpaperMigrationUtilityTests.swift

```diff
@@ -0,0 +1,160 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import XCTest
+import Shared
+
+@testable import Client
+
+class WallpaperMigrationUtilityTests: XCTestCase {
+
+    private let migrationKey = PrefsKeys.LegacyFeatureFlags.WallpaperDirectoryMigrationCheck
+
+    override func setUp() {
+        removeAllFolders()
+    }
+
+    override func tearDown() {
+        removeAllFolders()
+    }
+
+    func testRemovingAllFoldersOnSetup() {
+        verifyFoldersHaveBeenDeleted()
+    }
+
+    func testCreatingAFolder() {
+        verifyFoldersHaveBeenDeleted()
+
+        let fileManager = FileManager.default
+        guard let appSupportPath = path(for: .applicationSupport),
+              let docsPath = path(for: .documents)
+        else {
+            XCTFail("Could not create paths")
+            return
+        }
+
+        var isDirectory: ObjCBool = true
+        createFolderAt(path: docsPath)
+        createFolderAt(path: appSupportPath)
+
+        XCTAssertTrue(fileManager.fileExists(atPath: docsPath.path,
+                                              isDirectory: &isDirectory))
+        XCTAssertTrue(fileManager.fileExists(atPath: appSupportPath.path,
+                                              isDirectory: &isDirectory))
+
+    }
+
+    func testMigrationKeyDoesntExist() {
+        let profile = MockProfile(databasePrefix: "wallpaperMigrationTests")
+        XCTAssertNil(profile.prefs.boolForKey(migrationKey))
+    }
+
+    func testMigrationFlow() {
```

Comment:
What happens if there's no files? Should we have a test for that?

Reply:
Will add. :)

Reply:
I have updated with MOAR TESTSETSETSET!